lib/view_model/settings/mweb_settings_view_model.dart

```diff
@@ -22,15 +25,50 @@ abstract class MwebSettingsViewModelBase with Store {
   @observable
   late bool mwebEnabled;
 
+  @computed
+  String get mwebNodeUri => _settingsStore.mwebNodeUri;
+
   @action
   void setMwebCardDisplay(bool value) {
     _settingsStore.mwebCardDisplay = value;
   }
 
+  @action
+  void setMwebNodeUri(String value) {
+    _settingsStore.mwebNodeUri = value;
+  }
+
   @action
   void setMwebEnabled(bool value) {
     mwebEnabled = value;
     bitcoin!.setMwebEnabled(_wallet, value);
     _settingsStore.mwebAlwaysScan = value;
   }
+
+  Future<void> saveLogsLocally(String filePath) async {
+    final appSupportPath = (await getApplicationSupportDirectory()).path;
+    final logsFile = File("$appSupportPath/logs/debug.log");
+    if (!logsFile.existsSync()) {
+      throw Exception('Logs file does not exist');
```

Comment:
@fossephate this is not yet handled correctly, you only wrapped the places that call this function in a try and catch, but you missed some places that call the `share` function

why not just wrap this from the view model in a try and catch since you are using the exception handler now

Reply:
good call 👍